app/services/coupons/update_service.rb

```diff
@@ -60,6 +82,41 @@ def update_from_api(organization:, code:, params:)
 
     private
 
+    attr_reader :coupon, :limitations
+
+    def plan_identifiers
+      key = api_context? ? :plan_codes : :plan_ids
+      limitations[key]&.compact&.uniq
+    end
+
+    def plans
+      return @plans if defined? @plans
+      return [] if plan_identifiers.blank?
+
+      finder  = api_context? ? :code : :id
```

Comment:
Layout/SpaceAroundOperators: Operator `=` should be surrounded by a single space. (https://rubystyle.guide#spaces-operators)